common/metrics/simplereporter.go

```diff
@@ -36,6 +40,11 @@ type (
 	}
 )
 
+type HandlerFn func(metricName string, baseTags, tags map[string]string, value int64)
+
+var handlers = make(map[string]map[string]HandlerFn) // Key1 - metricName; Key2 - "filterTag:filterVal"
+var handlerMutex sync.RWMutex
+
```

Comment:
This is entirely by-design. Since we make the parent reporters as part of test setup, and then the product code makes sub-reporters and sub-sub-reporters, associating the handlers would require either defining the handler during test setup (and propagating to all children) or  having additional test hooks in the product to get a reference to particular child reporter.